subprojects/docs/src/docs/userguide/troubleshooting/validation_problems.adoc

```diff
@@ -389,3 +389,10 @@ To fix this, look at the error message which indicates the list of possible solu
 Please find below the list of unsupported value type and links to respective sections of the user manual:
 
 `ResolvedArtifactResult`:: <<incremental_build#sec:task_input_using_dependency_resolution_results, Using dependency resolution results as task inputs>>
+
+[[unsupported_key_type_of_nested_map]]
+== Key of nested map must be of type String
+
+This error indicates that a nested map declares a key of an unsupported type.
+
+To fix this problem, change the type of the key to String.
```

Comment:
🤔 I wonder if we should provide more context here. The user will be redirected here from the error message that already contains all the information we are presenting in this section. I think people will head here primarily when they have an `Integer` key; we could help them understand why we want them to use a `String` instead. I think a good explanation of that involves how Gradle's use of the key is to generate a name for the (sub)property, and we want to make sure the names are unique and well-formed instead of relying on `toString()` producing such names.